notebooks/learning-to-rank/01-learning-to-rank.ipynb

```diff
@@ -0,0 +1,1513 @@
+    "      }\n",
+    "   },\n",
+    "   \"rescore\" : {\n",
+    "      \"window_size\" : 50,\n",
+    "      \"learning_to_rank\" : {\n",
+    "         \"model_id\": \"ltr-model-xgboost\",\n",
+    "         \"params\": { \n",
+    "            \"query\": \"star wars\"\n",
+    "         }\n",
+    "      }\n",
+    "   }\n",
+    "}\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 145,
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/"
+    },
+    "id": "Xgr5MWWIrEk9",
+    "outputId": "e296cf37-afd1-43fb-e839-6c65cb65c072"
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "[('Star Wars', 10.972473, '11'),\n",
+       " ('Star Wars: The Clone Wars', 9.924128, '12180'),\n",
+       " ('After Porn Ends 2', 9.613241, '440249'),\n",
+       " ('Andor: A Disney+ Day Special Look', 8.982841, '1022100'),\n",
+       " (\"Family Guy Presents: It's a Trap!\", 8.840657, '278427'),\n",
+       " ('Star Wars: The Rise of Skywalker', 8.053794, '181812'),\n",
+       " ('Star Wars: The Force Awakens', 8.053794, '140607'),\n",
+       " ('Star Wars: The Last Jedi', 8.053794, '181808'),\n",
+       " ('Solo: A Star Wars Story', 8.053794, '348350'),\n",
+       " ('The Star Wars Holiday Special', 8.053794, '74849')]"
+      ]
+     },
+     "execution_count": 145,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "query = \"star wars\"\n",
+    "\n",
+    "# First let's display the result when not using the rescorer:\n",
+    "search_fields = [\"title\", \"overview\", \"actors\", \"director\", \"tags\", \"characters\"]\n",
+    "bm25_query = { \"multi_match\": { \"query\": query, \"fields\": search_fields } }\n",
+    "\n",
+    "bm25_search_response = es_client.search(index=MOVIE_INDEX, query=bm25_query)\n",
+    "\n",
+    "[\n",
+    "    (movie[\"_source\"][\"title\"], movie[\"_score\"], movie[\"_id\"])\n",
+    "    for movie in bm25_search_response['hits']['hits']\n",
+    "]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 146,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "[('Star Wars', 4.249656, '11'),\n",
+       " ('Star Wars: The Clone Wars', 3.037613, '12180'),\n",
+       " ('Star Wars: Episode I - The Phantom Menace', 1.7935627, '1893'),\n",
+       " ('Star Wars: The Force Awakens', 1.7227688, '140607'),\n",
+       " ('Star Wars: The Rise of Skywalker', 1.7130604, '181812'),\n",
+       " (\"Doraemon: Nobita's Little Star Wars 2021\", 1.6507263, '782054'),\n",
+       " ('Star Wars: Episode III - Revenge of the Sith', 1.5433255, '1895'),\n",
+       " ('Star Wars: Episode II - Attack of the Clones', 1.4986508, '1894'),\n",
+       " ('Star Wars: The Last Jedi', 1.4257026, '181808'),\n",
+       " ('Solo: A Star Wars Story', 1.0937448, '348350')]"
```

Comment:
it could be nice to round off the notebook with an observation on how the results changed due to rescoring, for example

> We saw above that the title and popularity fields are important ranking feature in our model. Here we can see that now all results contain the query terms in the title. Moreover, more popular movies rank higher, for example `Star Wars: Episode I - The Phantom Menace` is now in third position.